### ISSUE TITLE
How do I set up the path to mmgtools on Windows?

### ISSUE THREAD

**Issue by tin2tin** [norgeotloic/BakeMyScan#3]
There is a Windows build here of mmgtools:
https://forum.mmgtools.org/t/windows-compilation-succed-with-visualstudio/184/11

How do I set up the path to mmgtools in your Blender add-on?

**Comment by tin2tin** [comment]
Tried this in fn_soft.py, but it didn't work:
```
mmgsExe = "mmgs_O3 " if (platform.system() == "Linux" or platform.system() == "Darwin") else None

if platform.system() == "Windows":
    convertExe = "magick.exe convert "
    montageExe = "magick.exe montage "
    composeExe = "magick.exe composite "
    mmgsExe = "C:\\Users\\User\\Downloads\\bin\\bin\\mmgs_O3.exe "
```

**Comment by loicNorgeot** [comment]
Short: I have close to no idea, just specifying the path should have worked.
Longer: I'll try to have a look in the coming days as I don't have much time right now. If I manage to make it work correctly, I guess that I'll "externalize" the mmgs add-on, which could benefit from an add-on by itself.
Thanks for your interests, let me know if you find other stuff not working (which most probably will be the case, especially around the material imports and baking operations!)

**Comment by norgeotloic** [comment]
Short: I have close to no idea, just specifying the path should have worked.
Longer: I'll try to have a look in the coming days as I don't have much time right now. If I manage to make it work correctly, I guess that I'll "externalize" the mmgs add-on, which could benefit from an add-on by itself.
Thanks for your interests, let me know if you find other stuff not working (which most probably will be the case, especially around the material imports and baking operations!)

**Comment by tin2tin** [comment]
The Windows mmgtools are now available on the mmgtools download page: http://www.mmgtools.org/mmg-remesher-downloads
Maybe it is faster just to make the paths selectable in the UI of the add-on properties? (I'm not sure it works on Windows finding paths to installed apps in the registry? I have tried with other add-ons, but unsuccessfully )

**Comment by tin2tin** [comment]
Most options are greyed out. I get this error when hitting the MMGS button:
![image](https://user-images.githubusercontent.com/1322593/46850282-a1cb4380-cdf3-11e8-9fee-ce8cb20c376d.png)

I wonder if the temp folder is missing or where it is generated, so I can check if it is existing?

**Comment by norgeotloic** [comment]
Hi again.
I had a quick look, and indeed managed to make mmgs work on Windows.
Your suggestion to make the executable paths selectable in the user preferences is indeed the good one and I'll modify the add-on tonight/this weekend as I'm short on time right now.
The error message seems to be caused by blender trying to write tmp.mesh in the folder where it is executed (which is protected on Windows, but should work if you launch blender or blender.exe from the Powershell). 
For a quick fix, you could try replacing "tmp.mesh" with something such as "C:\\Users\\User\\Desktop" for instance. 
I'll adress this soon. Cheers!

**Comment by norgeotloic** [comment]
Hi again!
Concerning the error message you got on windows, it is due to the fact that mmgs does not output an error code of 0 on successfull completion.
Could you try installing [this .zip addon](https://github.com/norgeotloic/BakeMyScan/archive/dev.zip), which comes from the "dev" branch, and telling me if the fix works for you? (No need to modify the "advanced" options in the mmgs modal.).
Thanks!

**Comment by tin2tin** [comment]
Wow, so much you have done with the UI. It's really super cool.

Unfortunately all four remeshers are giving me errors:
[image: image.png]

Btw. in the preferences there are no options to add paths to the executive
files? Is this the reason to those errors?

Den søn. 28. okt. 2018 kl. 15.34 skrev Loïc Norgeot <
notifications@github.com>:

**Comment by norgeotloic** [comment]
OK, the zipping method I used was not the good one, and having a file called BakeMyScan-dev.zip instead of BakeMyScan.zip does not go well with the user preferences... 
That was most probably the reason to the errors (although I did not see the picture you attached).
I hope that finally [this .zip](https://github.com/norgeotloic/BakeMyScan/releases/download/0.2/BakeMyScan.zip) should work for you! (it worked for me between linux and windows).

**Comment by tin2tin** [comment]
Yes, the executive files now works(on Windows)!! :-) Looking forward to test all the rest of the functions, if that's okay with you?
